### DISCUSSION TITLE
gsoc is accepting organisations now!

### DISCUSSION THREAD

**Discussion by LingarajVB** [scalameta/metals#3630] [General]
Hello,
I was wondering, since Google summer of code is going to start in the next few months, and open source contributions will be funded, why can't metals apply? Is it even planned if it isn't already? If it isn't, now is the best time to do so.

Thank you,
Lingaraj

**Comment by LingarajVB** [comment]
@tgodzik @dos65 Can you guys check this ?

**Reply by kpodsiad** [reply]
It's a great idea! Applying to gsoc can also act as a catalyst for improving documentation, especially contributing guide. I'm all for it.

**Comment by tgodzik** [comment]
Thanks for the question! We are actually taking part in Google Summer of Code as part of Scala Center and there will be a couple of projects that can be done within Metals. We can use this thread for coordination, so that if you have any questions about the projects or need help do let us know!

**Reply by kpodsiad** [reply]
Is there a place where can I find more information on Scala Center participation in GSoC?

**Reply by tgodzik** [reply]
There should be an announcement coming soon on Scala twitter profile, that should have some more details.

From my side I added:
- semantic highlighting https://github.com/scalameta/metals-feature-requests/issues/57
- notebook support https://github.com/scalameta/metals-feature-requests/issues/236
- call hierarchy https://github.com/scalameta/metals-feature-requests/issues/188

There should also be one issue related to debugging, but the full list should be available once everything is set up and Scala Center is accepted to GSoC

**Reply by tgodzik** [reply]
Here is the repo we set up with ideas for this year https://github.com/scalacenter/GoogleSummerOfCode2022
